### PR TITLE
Hb 01

### PR DESCRIPTION
**Pull request by Alex-Korzun**
Don't mind on checks. I made a mistake. I think I fixed it but this pr should pass them anyway.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,6 +8,28 @@
...
+            <artifactId>hibernate-java8</artifactId>
+            <version>5.4.27.Final</version>
+        </dependency>
+
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.23</version>
+        </dependency>
+
```
Comment (Likh-Alex) [inline pom.xml:29]: redundant empty line

File: src/main/resources/hibernate.cfg.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        <property name ="hibernate.connection.driver_class">com.mysql.jdbc.Driver</property>
+        <property name ="hibernate.connection.url">jdbc:mysql://localhost/cinema?serverTimezone=UTC</property>
+        <property name ="hibernate.connection.username">root</property>
+        <property name ="hibernate.connection.password">1234</property>
+        <property name="show_sql">true</property>
+        <property name="hbm2ddl.auto">create-drop</property>
+
+        <mapping class="com.dev.cinema.model.Movie"/>
+    </session-factory>
+</hibernate-configuration>
```
Comment (Likh-Alex) [inline src/main/resources/hibernate.cfg.xml]: No new line at end of file

**Review by Likh-Alex — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (Alex-Korzun) [reply pom.xml:29]: removed

File: src/main/resources/hibernate.cfg.xml (hunk 2 above)
Comment (Alex-Korzun) [reply src/main/resources/hibernate.cfg.xml]: added

**Review by Likh-Alex — APPROVED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    @Override
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            Long itemId = (Long) session.save(movie);
+            transaction.commit();
+            movie.setId(itemId);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_01.html

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert Movie entity", e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't insert Movie entity " + movie, e);
```

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        } finally {
+            session.close();
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: let's add check for `null` value

File: src/main/java/com/dev/cinema/util/HibernateUtil.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    private static SessionFactory sessionFactory = initSessionFactory();
+
+    private HibernateUtil() {
+    }
+
+    private static SessionFactory initSessionFactory() {
+        try {
+            return new Configuration().configure().buildSessionFactory();
+        } catch (Exception e) {
+            throw new DataProcessingException("Error creating Session Factory", e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/util/HibernateUtil.java]: it isn't a data problem, so let's throw RuntimeExcpeption here

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java (hunk 3 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: removed setting id

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java (hunk 4 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: done

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java (hunk 5 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: done

File: src/main/java/com/dev/cinema/util/HibernateUtil.java (hunk 6 above)
Comment (Alex-Korzun) [reply src/main/java/com/dev/cinema/util/HibernateUtil.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**